Quadriga/src/main/java/edu/asu/spring/quadriga/web/projectblog/AddProjectBlogEntryController.java

```diff
@@ -132,15 +152,53 @@ public ModelAndView addProjectBlogEntryRequest(
         if (result.hasErrors()) {
             model = new ModelAndView("sites/addprojectblogentry");
             model.getModelMap().put("project", project);
+
         } else {
             String username = principal.getName();
             IUser user = userManager.getUser(username);
             projectBlogEntry.setAuthor(user);
 
             projectBlogEntryManager.addNewProjectBlogEntry(projectBlogEntry);
             model = new ModelAndView("redirect:" + "/sites/" + project.getUnixName() + "/projectblog");
+
             model.getModelMap().put("project", project);
         }
         return model;
     }
+
+    /**
+     * This method is used to return a JSON string for visualizing a network
+     * based on the Network id selected from the UI.
+     * 
+     * @param unixName
+     *            Unix Name given for the Project
+     * @param networkId
+     *            Network Id for the network selected from the UI.
+     * @param principal
+     *            principal object which is required to fetch information about
+     *            logged in user.
+     * @param project
+     *            project instance obtained using @InjectProject annotation
+     * @return Returns a JSON string as a response entity based on the network
+     *         selected from the UI.
+     * @throws QuadrigaStorageException
+     * @throws JAXBException
+     */
+    @AccessPolicies({ @ElementAccessPolicy(type = CheckedElementType.PROJECT, paramIndex = 5, userRole = {
```

Comment:
hmm, thinking about that again, this method is only used for creating a block entry, right? In that case this should work. If it's used to display networks anywhere else, then the authorization as it is done right now would be a problem. Let's add to the javadoc that this method works only for authorized users. We need to clean the network visualization up eventually, but for now this will probably work.